### PR TITLE
pkg: containerd now depends on kernel headers instead of libbtrfs

### LINKED ISSUE #87: btrfs headers issue with containerd for RHEL based distros
Same as https://github.com/docker/packaging/pull/69#issuecomment-1374731078 (fyi @neersighted)

https://github.com/docker/packaging/actions/runs/4158235007/jobs/7193316373#step:7:655

```
#41 52.20 + bin/containerd
#41 52.20 go build  -gcflags=-trimpath=/go/src -buildmode=pie  -o bin/containerd -ldflags '-X github.com/containerd/containerd/version.Version=v0.0.0-20230212032334-040fcf8 -X github.com/containerd/containerd/version.Revision=040fcf85f0a7d88ef3d55720d9821c9cb127c81c -X github.com/containerd/containerd/version.Package=containerd.io -s -w ' -tags "urfave_cli_no_docs"  ./cmd/containerd
#41 81.49 # github.com/containerd/btrfs/v2
#41 81.49 In file included from vendor/github.com/containerd/btrfs/v2/btrfs.go:21:0:
#41 81.49 ./btrfs.h:19:2: error: #error "Headers from kernel >= 4.12 are required on compilation time (not on run time)"
#41 81.49  #error "Headers from kernel >= 4.12 are required on compilation time (not on run time)"
#41 81.49   ^
#41 81.49 In file included from vendor/github.com/containerd/btrfs/v2/btrfs.go:21:0:
#41 81.49 ./btrfs.h:22:30: fatal error: linux/btrfs_tree.h: No such file or directory
#41 81.49  #include <linux/btrfs_tree.h>
```

### PR DESCRIPTION
**Pull request by crazy-max**
fixes #87 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Review by neersighted — APPROVED** [review]

**Comment by thaJeztah** [comment]
We may have to make this conditional, as it applies to containerd 1.7 and up (not (yet) on 1.6?)

**Comment by thaJeztah** [comment]
> We may have to make this conditional, as it applies to containerd 1.7 and up (not (yet) on 1.6?)

Ignore me; I wasn't aware this repo used a separate branch for containerd 1.6 (containers-packaging tries to be "generic" to allow building any version)

**Comment by crazy-max** [comment]
> > We may have to make this conditional, as it applies to containerd 1.7 and up (not (yet) on 1.6?)
> 
> Ignore me; I wasn't aware this repo used a separate branch for containerd 1.6 (containers-packaging tries to be "generic" to allow building any version)

Yes I also would prefer not to create release branches. I think your changes https://github.com/docker/containerd-packaging/pull/309/files look better in this case.

File: pkg/containerd/scripts/pkg-rpm-build.sh [hunk 1]
```diff
@@ -66,6 +66,12 @@ if [ -n "$(xx-info variant)" ]; then
   pkgoutput="${pkgoutput}/$(xx-info variant)"
 fi
 
+case "$PKG_RELEASE" in
+  centos7|centos8|centos9|oraclelinux*)
```
Comment (thaJeztah) [inline pkg/containerd/scripts/pkg-rpm-build.sh:70]: Do we need `rhel*` here as well?

I guess `centos*` (any version) would work for this going forward (we kept btrfs for 1.6 for now in containerd-packaging, to not drop it in a patch release)

Comment (crazy-max) [reply pkg/containerd/scripts/pkg-rpm-build.sh:70]: Ah right `centos*` would be enough. We don't publish specific `rhel` packages atm, only the following: https://github.com/docker/packaging/blob/55b6e3984539bf162dabdb697421b062c6840693/common/packages.mk#L19

Comment (thaJeztah) [reply pkg/containerd/scripts/pkg-rpm-build.sh:70]: Yeah, we don't build for RHEL ourselves, but I know IBM builds RHEL packages (for s390x and/or ppc64le), so was wondering if we need to account for that

Comment (neersighted) [reply pkg/containerd/scripts/pkg-rpm-build.sh:70]: I think it's fine for now, at some point when I get to porting the Mirantis build matrix over explicit RHEL might make sense, but it seems easy enough to grep for until then.

**Pull request merged**